File: blog/ros/beginner-guide-2/talker.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

#include <sstream>

int main(int argc, char **argv)
{
    // init with a name
    ros::init(argc, argv, "talker");

    // create a node
    ros::NodeHandle n;

    // publish on a the `chatter` topic, queue = 1000
    ros::Publisher chatter_pub = n.advertise<std_msgs::String>(
        "chatter",
        1000
    );

    // publishing rate 1 Hz
    ros::Rate loop_rate(1);

    // main loop
    int count = 0;
    while (ros::ok()) {
        // message
        std_msgs::String msg;

        // content
        std::stringstream ss;
        ss << "hello world " << count++;
        msg.data = ss.str();
        ROS_INFO("%s", msg.data.c_str());

        // publish
        chatter_pub.publish(msg);

        // check status
        ros::spinOnce();

        // sleep
        loop_rate.sleep();
    }

    return 0;
}

File: blog/ros/beginner-guide-2/talker.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String

def talker():
    # create a publisher, on topic `chatter`
    pub = rospy.Publisher('chatter', String, queue_size=10)

    # create a node
    rospy.init_node('talker', anonymous=True)

    # set the rate of publishing
    rate = rospy.Rate(1) # 1hz

    # main loop
    while not rospy.is_shutdown():
        # make content
        hello_str = "hello world %s" % rospy.get_time()
        rospy.loginfo(hello_str)

        # publish a message
        pub.publish(hello_str)

        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass
